### ISSUE TITLE
Router panic when trying to update route

### ISSUE THREAD

**Issue by germancho14** [fission/fission#810]
On fission 0.9.1 I've noticed the following error in the Router Pod:
```
.
.
.
2018/07/19 22:43:55 Skipping creation of ingress for trigger: d8f1d53f-9831-4faf-8641-6ce7241df698
2018/07/19 22:43:57 Calling getServiceForFunction for function: fn-96a377a1-8dc9-4f2b-96a7-c370fc75d478
2018/07/19 22:43:58 Skipping creation of ingress for trigger: 2276415b-33c7-44e4-acc0-273d46b0878c
E0719 22:43:58.499186       1 runtime.go:66] Observed a panic: "index out of range" (runtime error: index out of range)
/go/src/github.com/fission/fission/vendor/k8s.io/apimachinery/pkg/util/runtime/runtime.go:72
/go/src/github.com/fission/fission/vendor/k8s.io/apimachinery/pkg/util/runtime/runtime.go:65
/go/src/github.com/fission/fission/vendor/k8s.io/apimachinery/pkg/util/runtime/runtime.go:51
/usr/local/go/src/runtime/asm_amd64.s:573
/usr/local/go/src/runtime/panic.go:502
/usr/local/go/src/runtime/panic.go:28
/go/src/github.com/fission/fission/router/httpTriggers.go:105
/go/src/github.com/fission/fission/router/httpTriggers.go:251
/go/src/github.com/fission/fission/router/httpTriggers.go:221
/go/src/github.com/fission/fission/vendor/k8s.io/client-go/tools/cache/controller.go:202
/go/src/github.com/fission/fission/vendor/k8s.io/client-go/tools/cache/controller.go:309
/go/src/github.com/fission/fission/vendor/k8s.io/client-go/tools/cache/delta_fifo.go:444
/go/src/github.com/fission/fission/vendor/k8s.io/client-go/tools/cache/delta_fifo.go:444
/go/src/github.com/fission/fission/vendor/k8s.io/client-go/tools/cache/controller.go:150
/go/src/github.com/fission/fission/vendor/k8s.io/client-go/tools/cache/controller.go:124
/go/src/github.com/fission/fission/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:133
/go/src/github.com/fission/fission/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:134
/go/src/github.com/fission/fission/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:88
/go/src/github.com/fission/fission/vendor/k8s.io/client-go/tools/cache/controller.go:124
/go/src/github.com/fission/fission/router/httpTriggers.go:229
/usr/local/go/src/runtime/asm_amd64.s:2361
.
.
.
```

The pod is not crashing, but I believe this is causing the Informer to stop sending notifications on changes on Fission's Functions and Triggers. As a result the router pod doesn't know about new Triggers and Functions and returns 404 when a client tries to invoke one. Once I kill the router pod (and then get a new one). I stop getting 404s and requests get routed correctly. Which I guess means that the router knows about all the new routes because it resyncs at startup.

**Comment by life1347** [comment]
@germancho14 
I think the problem is caused by `/go/src/github.com/fission/fission/router/httpTriggers.go:105` due to the size of ts.triggers was changed and caused this error. We will fix this.
And really thanks for your reporting! :)

**Comment by soamvasani** [comment]
Fixed by #811
